episodes/episode-module.nix

```
@@ -0,0 +1,145 @@
+{
+  lib,
+  options,
+  config,
+  episode_dir,
+  previous_episode,
+  ...
+}: {
+  options = {
+    episode_number = lib.mkOption {
+      type = lib.types.str;
+    };
+
+    youtube_id = lib.mkOption ({
+        type = lib.types.str;
+      }
+      // lib.optionalAttrs (builtins.pathExists (episode_dir + "/youtube_id")) {
+        default = lib.removeSuffix "\n" (builtins.readFile (episode_dir + "/youtube_id"));
+      });
```

```suggestion
    youtube_id = lib.mkOption ({
        type = lib.types.nullOr lib.types.str;
        default =
          if builtins.pathExists (episode_dir + "/youtube_id") then
            lib.fileContents (episode_dir + "/youtube_id")
          else
            null;
      };
```

Generally it's best to make the type as accurate as possible